### PR TITLE
added ShoppingCart tests

### PR DESCRIPTION
**Pull request by RCherniavskiy**

### CODE REVIEW

File: src/test/java/book_store/controller/ShoppingCartControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        mockMvc.perform(get("/shoppingCarts").with(authentication(authentication)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(1))
+                .andExpect(jsonPath("$.cartItems.length()").value(2));
+    }
+
+    @Test
+    @DisplayName("Add book to shopping cart")
+    void addBookToShoppingCart() throws Exception {
```
Comment (koval-kn) [inline src/test/java/book_store/controller/ShoppingCartControllerTest.java]: test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/book_store/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    @MockBean
+    private ShoppingCartService shoppingCartService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @DisplayName("Get shopping cart")
+    void getShoppingCart() throws Exception {
```
Comment (koval-kn) [inline src/test/java/book_store/controller/ShoppingCartControllerTest.java]: test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/book_store/controller/ShoppingCartControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                        .with(authentication(authentication)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(1))
+                .andExpect(jsonPath("$.cartItems.length()").value(2));
+    }
+
+    @Test
+    @DisplayName("Update the quantity of goods in the cart")
+    void updateCartItemQuantity() throws Exception {
```
Comment (koval-kn) [inline src/test/java/book_store/controller/ShoppingCartControllerTest.java]: test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/book_store/service/ShoppingCartServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    private BookMapper bookMapper;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    @DisplayName("Test findUserShoppingCart: Success - User Exists, ShoppingCart Returned")
+    void testFindUserShoppingCart_Success_UserExists_ShoppingCartReturned() {
```
Comment (koval-kn) [inline src/test/java/book_store/service/ShoppingCartServiceImplTest.java]: test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/book_store/service/ShoppingCartServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        when(shoppingCartRepository.findByUserId(user.getId(), pageable)).thenReturn(page);
+        ShoppingCartDto expectedDto = new ShoppingCartDto();
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(expectedDto);
+        ShoppingCartDto resultDto = shoppingCartService.findUserShoppingCart(user, pageable);
+        assertEquals(expectedDto, resultDto);
+    }
+
+    @Test
+    @DisplayName("Test addBookToCart: Success - Book Added To Cart, CartItem Quantity Updated")
+    void testAddBookToCart_Success_BookAddedToCart_CartItemQuantityUpdated() {
```
Comment (koval-kn) [inline src/test/java/book_store/service/ShoppingCartServiceImplTest.java]: test name invalid, should be in pattern like:
`<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/book_store/service/ShoppingCartServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        when(cartItemRepository.findByShoppingCartAndBook(shoppingCart, book)).thenReturn(existingCartItem);
+        when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(shoppingCartDto);
+        ShoppingCartDto result = shoppingCartService.addBookToCart(request, user);
+        assertEquals(shoppingCartDto, result);
+        assertEquals(2, existingCartItem.getQuantity());
+    }
+
+    @Test
+    @DisplayName("Test updateCartItemQuantity: Success")
+    public void testUpdateCartItemQuantity_Success() {
```
Comment (koval-kn) [inline src/test/java/book_store/service/ShoppingCartServiceImplTest.java]: `test` in name is redundant, fix in all project

**Review by koval-kn — CHANGES_REQUESTED** [review]
not implemented:
- Please add a screenshot with test coverage results into the PR description
- Add test cases for ShoppingCart entity (*repository, controller*)

**Comment by RCherniavskiy** [comment]
tests

**Review by sarakhmen — APPROVED** [review]
Good job !

**Pull request merged**